app/src/main/java/com/example/bluetoothdevicesapplication/ui/view/DevicesFragment.kt

```diff
@@ -1,38 +1,71 @@
 package com.example.bluetoothdevicesapplication.ui.view
 
+import android.bluetooth.BluetoothAdapter
+import android.bluetooth.BluetoothDevice
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
 import androidx.lifecycle.ViewModelProvider
 import android.os.Bundle
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.navigation.Navigation
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.bluetoothdevicesapplication.R
 import com.example.bluetoothdevicesapplication.ui.viewmodel.DevicesViewModel
-import kotlinx.android.synthetic.main.devices_fragment.view.*
+import kotlinx.android.synthetic.main.devices_fragment.*
 
 class DevicesFragment : Fragment() {
 
-    companion object {
-        fun newInstance() = DevicesFragment()
-    }
-
     private lateinit var viewModel: DevicesViewModel
+    private lateinit var broadcastReceiver: BroadcastReceiver
+    private val intentFilter = IntentFilter()
+
+    init {
+        intentFilter.addAction(BluetoothDevice.ACTION_BOND_STATE_CHANGED)
+        intentFilter.addAction(BluetoothDevice.ACTION_ACL_CONNECTED)
+        intentFilter.addAction(BluetoothDevice.ACTION_ACL_DISCONNECTED)
+        intentFilter.addAction(BluetoothAdapter.ACTION_STATE_CHANGED)
+    }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val view = inflater.inflate(R.layout.devices_fragment, container, false)
-        view.textViewDevices.setOnClickListener { Navigation.findNavController(view).navigate(R.id.action_devicesFragment_to_detailFragment) }
-
-        return view
+        return inflater.inflate(R.layout.devices_fragment, container, false)
     }
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         viewModel = ViewModelProvider(this).get(DevicesViewModel::class.java)
-        // TODO: Use the ViewModel
+        viewModel.pairedDevices.observe(viewLifecycleOwner, {
```

Comment:
Very good, the lifecycle concept in Android is important, helps you prevent leaks and keeps your app performant